tenant/configloader/loader.go

```diff
@@ -0,0 +1,112 @@
+package configloader
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/oauth2-proxy/oauth2-proxy/v7/providers"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/tenant/types"
+	"github.com/spf13/viper"
+)
+
+type loader struct {
+	conf        *Configuration           // tenant loader configuration
+	tenantsConf *TenantsConfig           // tenants configuration that has been loaded from file at path loader.conf.TenantsFile
+	tenants     map[string]*types.Tenant // tenants map, key is tenant id
+
+	rules []*rule
+}
+
+func New(conf *Configuration) (*loader, error) {
+	tntsConf := &TenantsConfig{}
+
+	err := loadConfigFromFile(conf.TenantsFile, tntsConf)
```

Comment:
Sharding of config files is just going to make our configuration story harder IMO

Reply:
will merge the config files,

Reply:
resolve, only once config file for now.